astronomer/providers/amazon/aws/triggers/s3.py

```diff
@@ -18,95 +16,189 @@ def __init__(
+                        )
+
+                        if self.check_fn_user is not None:
+                            if self.check_fn_user(s3_objects):
+                                yield TriggerEvent({"status": "success"})
+                                return
+
+                        if self._check_fn(s3_objects):
+                            yield TriggerEvent({"status": "success"})
+                            return
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
 
     def _get_async_hook(self) -> S3HookAsync:
         return S3HookAsync(aws_conn_id=self.aws_conn_id, verify=self.hook_params.get("verify"))
+
+
+class S3KeysUnchangedTrigger(BaseTrigger):
+    def __init__(
+        self,
+        bucket_name: str,
+        prefix: str,
+        inactivity_period: float = 60 * 60,
+        min_objects: int = 1,
+        inactivity_seconds: int = 0,
+        previous_objects: Optional[Set[str]] = None,
+        allow_delete: bool = True,
+        aws_conn_id: str = "aws_default",
+        last_activity_time: Optional[datetime] = None,
+        verify: Optional[Union[bool, str]] = None,
+    ):
+        self.bucket_name = bucket_name
+        self.prefix = prefix
+        if inactivity_period < 0:
+            raise ValueError("inactivity_period must be non-negative")
+        self.inactivity_period = inactivity_period
+        self.min_objects = min_objects
+        self.previous_objects = previous_objects or set()
+        self.inactivity_seconds = inactivity_seconds
+        self.allow_delete = allow_delete
+        self.aws_conn_id = aws_conn_id
+        self.last_activity_time: Optional[datetime] = last_activity_time
+        self.verify = verify
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """
+        Serialize S3KeysUnchangedTrigger arguments and classpath.
+        """
+        return (
+            "astronomer.providers.amazon.aws.triggers.s3.S3KeysUnchangedTrigger",
+            {
+                "bucket_name": self.bucket_name,
+                "prefix": self.prefix,
+                "inactivity_period": self.inactivity_period,
+                "min_objects": self.min_objects,
+                "previous_objects": self.previous_objects,
+                "inactivity_seconds": self.inactivity_seconds,
+                "allow_delete": self.allow_delete,
+                "aws_conn_id": self.aws_conn_id,
+                "last_activity_time": self.last_activity_time,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """
+        Make an asynchronous connection using S3HookAsync.
+        """
+        try:
+            hook = self._get_async_hook()
+            async with await hook.get_client_async() as client:
+                while True:
+
+                    if await hook.is_keys_unchanged(
```

Comment:
These lines not covered by tests

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/86/commits/8fcb589e89f627cf908b89c7254298c2ece9a09f)